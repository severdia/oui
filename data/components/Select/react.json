{"description":"Generates an <select> element","displayName":"Select","methods":[],"props":{"children":{"type":{"name":"node"},"required":false,"description":"options elements"},"id":{"type":{"name":"string"},"required":false,"description":"select id"},"isDisabled":{"type":{"name":"bool"},"required":false,"description":"boolean for when select is disabled"},"name":{"type":{"name":"string"},"required":false,"description":"select name"}},"private":false}