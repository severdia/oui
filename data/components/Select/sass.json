{"name":"Select\n","description":"","start":{"line":1,"column":1},"end":{"line":13,"column":4},"example":[{"type":"html","description":null,"code":"<select name=\"zoo\" id=\"zoo\" class=\"#{$namespace}select\">\n  <option value=\"one\">This is option one</option>\n  <option value=\"two\">And this is option two</option>\n</select>"},{"type":"html","description":null,"code":"<select name=\"zoo\" disabled id=\"zoo\" class=\"#{$namespace}select\">\n  <option value=\"one\">This is option one</option>\n  <option value=\"two\">And this is option two</option>\n</select>"}]}