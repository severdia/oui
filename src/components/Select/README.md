# Select Component

- Add storybook link
- Add deisgn guidelines