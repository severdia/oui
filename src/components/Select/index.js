import React from 'react';
import PropTypes from 'prop-types';

/**
 * Generates an <select> element
 * @param {Object} props - Properties passed to component
 * @returns {ReactElement}
 */

const Select = ({
  children,
  id,
  isDisabled,
  name,
}) => {

  return (
    <select
      data-oui-component={ true }
      disabled={ isDisabled }
      name={ name }
      id={ id }
      className="oui-select">
      { children }
    </select>
  );
};

Select.propTypes = {
  /** options elements */
  children: PropTypes.node,
  /** select id */
  id: PropTypes.string,
  /** boolean for when select is disabled */
  isDisabled: PropTypes.bool,
  /** select name */
  name: PropTypes.string,
};

Select.displayName = 'Select';

export default Select;
