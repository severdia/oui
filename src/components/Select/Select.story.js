import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs, boolean } from '@storybook/addon-knobs';

import Select from './index.js';

const stories = storiesOf('Select', module);
stories
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div id="root-preview">
      {story()}
    </div>
  ));

stories.add('Default', (() => {
  return (
    <Select isDisabled={ boolean('isDisabled', false) } name='zoo' id='zoo' >
      <option value="one">This is option one</option>
      <option value="two">And this is option two</option>
    </Select>
  );
}));
