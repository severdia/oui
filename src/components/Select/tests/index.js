import React from 'react';
import Select from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('components/Select', () => {
  it('should render correctly when declared with no properties', () => {
    const output = shallow(
      <Select>
        <option value="one">This is option one</option>
        <option value="two">And this is option two</option>
      </Select>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render correctly when isDisabled prop is declare', () => {
    const output = shallow(
      <Select isDisabled={ true } >
        <option value="one">This is option one</option>
        <option value="two">And this is option two</option>
      </Select>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

});
