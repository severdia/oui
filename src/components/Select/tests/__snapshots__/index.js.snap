// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Select should render correctly when declared with no properties 1`] = `
<select
  className="oui-select"
  data-oui-component={true}
>
  <option
    value="one"
  >
    This is option one
  </option>
  <option
    value="two"
  >
    And this is option two
  </option>
</select>
`;

exports[`components/Select should render correctly when isDisabled prop is declare 1`] = `
<select
  className="oui-select"
  data-oui-component={true}
  disabled={true}
>
  <option
    value="one"
  >
    This is option one
  </option>
  <option
    value="two"
  >
    And this is option two
  </option>
</select>
`;
